python/lsst/pipe/base/connectionTypes.py

```diff
@@ -153,7 +154,32 @@ class Input(BaseInput):
 
 @dataclasses.dataclass(frozen=True)
 class PrerequisiteInput(BaseInput):
-    pass
+    """Class used for declaring PipelineTask prerequisite connections
+
+    Parameters
+    ----------
+    name : `str`
+        The name used to identify the dataset type
+    storageClass : `str`
+        The storage class used when (un)/persisting the dataset type
+    multiple : `bool`
+        Indicates if this connection should expect to contain multiple objects
+        of the given dataset type
+    dimensions : iterable of `str`
+        The `lsst.daf.butler.Butler` `lsst.daf.butler.Registry` dimensions used
+        to identify the dataset type identified by the specified name
+    deferLoad : `bool`
+        Indicates that this dataset type will be loaded as a
+        `lsst.daf.butler.DeferredDatasetHandle`. PipelineTasks can use this
+        object to load the object at a later time.
+    lookupFunction: `typing.Callable`
+        An optional callable function that will look up PrerequisiteInputs
+        using the DatasetType, registry, quantum dataId, and input collections
+        passed to it. If no funciton is specified, the default temporal spatial
+        lookup will be used.
+    """
+    lookupFunction: typing.Optional[typing.Callable[[DatasetType, Registry, ExpandedDataCoordinate, Mapping],
+                                    typing.Iterable[typing.Any]]] = None
```

Comment:
I dont have a strong opinion here. My only feeling is often these names are pretty generic so it may help people unfamiliar with typing to see where they come from. It may be my linter but this was the only position it was happy with so I left it. I will change the import and sees what happens.